java-squid/src/main/java/org/sonar/java/se/SymbolicValueRelation.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * SonarQube Java
+ * Copyright (C) 2012-2016 SonarSource SA
+ * mailto:contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.java.se;
+
+import javax.annotation.Nullable;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+public abstract class SymbolicValueRelation {
+
+  @Nullable
+  public static Boolean checkValidity(SymbolicValueRelation relation, Collection<SymbolicValueRelation> knownRelations) {
+    Boolean result = checkValidityNoInverse(relation, knownRelations);
+    if (result != null) {
+      return result;
+    }
+    Boolean inverseResult = checkValidityNoInverse(relation.inverse(), knownRelations);
+    if (inverseResult != null) {
+      return Boolean.valueOf(!inverseResult.booleanValue());
+    }
+    return null;
+  }
+
+  @Nullable
+  public static Boolean checkValidityNoInverse(SymbolicValueRelation relation, Collection<SymbolicValueRelation> knownRelations) {
+    SymbolicValueRelation symmetric = relation.symmetric();
+    for (SymbolicValueRelation otherRelation : knownRelations) {
+      Boolean validity = checkValidity(relation, otherRelation, knownRelations);
+      if (validity != null) {
+        return validity;
+      }
+      validity = checkValidity(symmetric, otherRelation, knownRelations);
+      if (validity != null) {
+        return validity;
+      }
+    }
+    return null;
+  }
+
+  @Nullable
+  private static Boolean checkValidity(SymbolicValueRelation rel1, SymbolicValueRelation rel2, Collection<SymbolicValueRelation> knownRelations) {
+    if (rel1.v1.equals(rel2.v1) && rel1.v2.equals(rel2.v2)) {
+      return rel1.consistentWith(rel2);
```

Comment:
This is the only usage of this method. As such it should be either made private or be inlined (I prefer the latter personnaly for this case and I'm wondering if that should be implemented with equals/hashcode on SVR ?)